### PR TITLE
Get the GetService into HttpContextWrapper

### PR DESCRIPTION
**Pull request by birojnayak**
Porting GetService  to HttpContextWrapper . This will fix the issue #456

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpContextWrapper.cs [hunk 1]
```diff
@@ -81,5 +81,10 @@ public override IPrincipal User
         public override void RewritePath(string filePath, string pathInfo, string? queryString, bool setClientFilePath) => _context.RewritePath(filePath, pathInfo, queryString, setClientFilePath);
 
         public override void SetSessionStateBehavior(SessionStateBehavior sessionStateBehavior) => _context.SetSessionStateBehavior(sessionStateBehavior);
+
+        public override object GetService(Type serviceType)
+        {
+            return ((HttpContextCore)_context).RequestServices.GetService(serviceType);
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpContextWrapper.cs:87]: this should delegate to the _context.GetService method. The `HttpContext.GetService()` method could hook into the _context.RequestServices

File: test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpContextTests.cs [hunk 2]
```diff
@@ -152,8 +153,6 @@ public void GetServiceReturnsExpected()
             Assert.Same(context.Response, provider.GetService(typeof(HttpResponse)));
             Assert.Same(context.Server, provider.GetService(typeof(HttpServerUtility)));
             Assert.Same(context.Session, provider.GetService(typeof(HttpSessionState)));
-
-            Assert.Null(provider.GetService(typeof(HttpContext)));
```
Comment (birojnayak) [inline test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpContextTests.cs:156]: The code coverage is achieved via CacheTest.cs.

Comment (twsouthwick) [reply test/Microsoft.AspNetCore.SystemWebAdapters.Tests/HttpContextTests.cs:156]: It makes more sense to add the tests to a HttpContextTests class as the CacheTest shouldn't actually be testing that

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpContextWrapper.cs [hunk 3]
```diff
@@ -81,5 +81,10 @@ public override IPrincipal User
         public override void RewritePath(string filePath, string pathInfo, string? queryString, bool setClientFilePath) => _context.RewritePath(filePath, pathInfo, queryString, setClientFilePath);
 
         public override void SetSessionStateBehavior(SessionStateBehavior sessionStateBehavior) => _context.SetSessionStateBehavior(sessionStateBehavior);
+
+        public override object? GetService(Type serviceType)
+        {
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpContextWrapper.cs:86]: nit: use expression instead similar to the rest of the methods

**Review by twsouthwick — CHANGES_REQUESTED** [review]

**Review by twsouthwick — APPROVED** [review]
LGTM

File: src/Microsoft.AspNetCore.SystemWebAdapters/HttpContext.cs [hunk 4]
```diff
@@ -162,7 +162,10 @@ public void RewritePath(string filePath, string pathInfo, string? queryString, b
         {
             return Server;
         }
-
+        else if (Context.RequestServices != null)
+        {
+            return Context.RequestServices.GetService(service);
+        }
         return null;
```
Comment (twsouthwick) [inline src/Microsoft.AspNetCore.SystemWebAdapters/HttpContext.cs:169]:
```suggestion
        return Context.RequestServices.GetService(service);
```

**Pull request merged**